statchatui/src/Compare/Compare.jsx

```diff
@@ -43,34 +37,45 @@ const Compare = ({ signedInUserData }) => {
     setIsLoading(true); //sets the state to true before the api call is displaying the stats
     let stats1;
     let stats2;
+    let validUser = true;
     try {
       stats1 = await fetchStats(user1);
       stats2 = await fetchStats(user2);
     } catch (error) {
       setError(error?.message);
+      validUser = false;
     }
+    //if there was an error fetching the stats( meaning that the user does not exist) then it will not be added to the user queue
+    if (validUser) {
+      setOtherUserStats(stats1);
+      setSignedInUserStats(stats2);
+      setOtherUserFormData(user1);
+      setSignedInUserFormData(user2);
 
-    setOtherUserStats(stats1);
-    setSignedInUserStats(stats2);
-    setOtherUserFormData(user1);
-    setSignedInUserFormData(user2);
+      //checks if the user is in the userqueue already
+      const isUserInQueue = userQueue.some(
+        (user) => user.gamertag === user1.gamertag
+      );
+      if (!isUserInQueue) {
+        const newUser = {
+          gamertag: user1.gamertag,
+          platform: user1.platform,
+          gameName: user1.gameName,
+          // Assigned the newUser to have the stats of stats1 because otherUser and newUser are the same and I want the stats to be saved in the userQueue
```

Comment:
I'm a little confused by this comment. Is it possible some of the variable names changed since you first wrote it?

Reply:
yes, this was for a different version of the code, thank you for pointing it out:


new comment:          //assigned the stats for the new user as the user1.stats to user later.